File: sorting/selection_sort/selection.rb
unsorted_array = [5, 3, 42, 398, 28, 2, 20, 5, 2, 0, -20, 4]

def selection_sort(a)
  n = a.size - 1

  n.times do |i|
    index_min = i

    (i + 1).upto(n) do |j|
      index_min = j if a[j] < a[index_min]
    end

   a[i], a[index_min] = a[index_min], a[i] if index_min != i
  end

  return a
end

puts selection_sort(unsorted_array)

File: sorting/selection_sort/selection.js
var unsorted_array = [5, 3, 42, 398, 28, 2, 20, 5, 2, 0, -20, 4, 2, 4];

var selectionSort = function (a) {
  var n = a.length;
  for (var i = 0; i < n - 1; i++) {
    var min = i;
    for (var j = i + 1; j < n; j++) {
      if (a[min] > a[j]) // Find the smallest item that needs to be swapped
        min = j;
    }
    if (min != i) {
      var tmp = a[i];
      a[i] = a[min];
      a[min] = tmp;
      console.log(a);
    }
  }

  return a;
};

result = selectionSort(unsorted_array);
console.log(result);
